### PR TITLE
feat(core-flows,order,medusa): exchange endpoints

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
Order Exchange: PR 2:n

What:
- Order Exchange endpoints
- Claims and Exchange worfklows fixes 
- Small fixes on Order module action validations

Next PR will contain inventory management of claims and exchanges.

### CODE REVIEW

File: packages/core/core-flows/src/order/workflows/exchange/remove-exchange-item-action.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        `No item exchange found for exchange ${input.exchange_id} in order change ${orderChange.id}`
+      )
+    } else if (associatedAction.action !== ChangeActionType.ITEM_ADD) {
+      throw new Error(`Action ${associatedAction.id} is not adding an item`)
+    }
+  }
+)
+
+export const removeItemExchangeActionWorkflowId = "remove-item-exchange-action"
+export const removeItemExchangeActionWorkflow = createWorkflow(
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/exchange/remove-exchange-item-action.ts:57]: would love to get these steps to be reusable soon 😄

File: packages/medusa/src/api/admin/exchanges/[id]/inbound/items/[action_id]/route.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import {
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/exchanges/[id]/inbound/items/[action_id]/route.ts:1]: question: Why do we fetch and return `returns` in these endpoints when they are for exchanges?

File: packages/medusa/src/api/admin/exchanges/[id]/inbound/items/route.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { orderExchangeRequestItemReturnWorkflow } from "@medusajs/core-flows"
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/exchanges/[id]/inbound/items/route.ts:1]: same question here

File: packages/medusa/src/api/admin/exchanges/[id]/inbound/shipping-method/[action_id]/route.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import {
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/exchanges/[id]/inbound/shipping-method/[action_id]/route.ts:1]: Same question here, will not comment more on this anymore if there are other similar endpoints.

**Review by olivermrbl — APPROVED** [review]
Lots of stuff in this PR. Can we try to keep them smaller going forward because it's a bit hard to review? 

I left a few comment, but I would be OK getting this merged so it doesn't block any UI implementation work.

File: packages/medusa/src/api/admin/exchanges/[id]/inbound/items/[action_id]/route.ts (hunk 2 above)
Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/admin/exchanges/[id]/inbound/items/[action_id]/route.ts:1]: In practice, inbounds are returns

File: packages/core/core-flows/src/order/workflows/exchange/remove-exchange-item-action.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/exchange/remove-exchange-item-action.ts:57]: I will work to clean these similar steps everywhere in Order flows

**Pull request merged**